File: src/Belajar_SOLID_05_OOP_Polymorphism/Java/Runtime_Polymorphism.java
package Belajar_SOLID_05_OOP_Polymorphism.Java;

class Animal{
    public void walk(){
        System.out.println(getClass().getSimpleName() + " walk");
    }
}

class Cat extends Animal{
    @Override
    public void walk() {
        super.walk();
        System.out.println(getClass().getSimpleName() + " walked...!");
    }
}

public class Runtime_Polymorphism {
    public static void main(String[] args) {
        Cat kucing = new Cat();
        kucing.walk();
    }
}

File: src/Belajar_SOLID_05_OOP_Polymorphism/Kotlin/Runtime_Polymorphism.kt
package Belajar_SOLID_05_OOP_Polymorphism.Kotlin

open class Animal{
    open fun walk(){
        println("${javaClass.simpleName} walk")
    }
}

class Cat:Animal(){
    // Fungsi dipanggil saat runtime contoh override
    override fun walk() {
        super.walk()
        println("Yeyy.. ${javaClass.simpleName} walked...!")
    }
}

fun main(){
    val cat = Cat()
    cat.walk()
}